src/array/list/mutable.rs

```diff
@@ -181,6 +181,7 @@ impl<O: Offset, M: MutableArray> MutableListArray<O, M> {
     /// - the new offsets are not in monotonic increasing order.
     /// - any new offset is not in bounds of the backing array.
     /// - the passed iterator has no upper bound.
+    #[allow(dead_code)]
```

Comment:
I'm not sure do we need `extend_offsets` and `unsafe_extend_offsets` anymore.

Cc @jorgecarleitao for comments.

Reply:
I believe that this is only used when json feature is active.

Reply:
Since clippy didn't test with the `json` feature enabled, It's OK to add a `#[allow(dead_code)]` here?